nemo_skills/evaluation/metrics/base.py

```diff
@@ -13,29 +13,281 @@
+        k: int,
+        score_method: str,
+        score_dicts: list[dict],
+        majority_score: bool | float | int,
+        majority_answer: str,
+        predictions: list[dict],
+        predicted_answers: list[str],
+    ):
+        """
+        Update the metrics dictionary with additional statistics.
+
+        Called by `_compute_majority_at_k` in case there are other metrics we want to log.
+
+        This method is being called in a loop for each score_method, so only
+        use it for metrics that depend on the correctness method.
+        """
+
+    def _update_metrics_for_majority(
+        self,
+        eval_dict: dict,
+        k: int,
+        predictions: list[dict],
+        predicted_answers: list[str],
+    ):
+        """
+        Update the metrics dictionary with additional statistics.
+
+        Called by `_compute_pass_at_k` in case there are other metrics we want to log.
+
+        Unlike `_update_score_metrics_for_pass`, this method is called one time after the
+        loop over all `score_method` in `_compute_pass_at_k`.
+
+        It can be used for metrics that do not depend on the correctness method.
+        """
+
+    def _compute_majority_at_k(
+        self, predictions: list[dict], predicted_answers: list[str], eval_dict: dict | None = None
+    ):
+        """
+        Get majority@k metrics for a given set of prediction results.
+
+        Args:
+            predictions (list): List of generated predictions.
+                Will call `_get_score_dict` to get scores for predictions.
+            predicted_answers (list): List of the answers that we should use to compute majority.
+            eval_dict (Optional[dict]): Dictionary to store aggregated metrics.
+                By default will use self.eval_dict.
+        """
+        if eval_dict is None:
+            eval_dict = self.eval_dict
+
+        score_dicts = [self._get_score_dict(pred) for pred in predictions]
+
+        for k in range(2, len(predictions) + 1):
+            for score_method in score_dicts[0].keys():
+                # Get valid answers and their results for this field
+                valid_answers_and_results = [
+                    (pred_answer, correctness_dict[score_method])
+                    for pred_answer, correctness_dict in zip(predicted_answers[:k], score_dicts[:k])
+                    if pred_answer is not None
+                ]
+
+                # If no valid answers, it's incorrect
+                if not valid_answers_and_results:
+                    majority_score = 0
+                    majority_answer = None
+                else:
+                    # sorting to ensure reproducible scores in case of ties in majority
+                    valid_answers_and_results = sorted(valid_answers_and_results)
+                    # Find the most common answer and its correctness
+                    majority_answer, majority_score = Counter(valid_answers_and_results).most_common(1)[0][0]
```

Comment:
Need a tiebreak logic here. Maybe allow for tiebreaks if a tiebreak method is implemented.